core/storage/base_model/gae_models.py

```diff
@@ -84,47 +84,32 @@
 
 
 class BaseModel(datastore_services.Model):
-    """Base model for all persistent object storage classes."""
+    """Base model for all persistent object storage classes.
+
+    IMPORTANT: If humans can interface with the model directly, then inherit
+    from BaseHumanMaintainedModel instead and use the put_for_human family of
+    methods when appropriate.
+    """
 
     # Specifies whether the model's id is used as a key in Takeout. By default,
     # the model's id is not used as the key for the Takeout dict.
     ID_IS_USED_AS_TAKEOUT_KEY = False
 
-    # When this entity was first created. This value should only be modified by
-    # the update_timestamps method.
+    # When this entity was first created. This value should not be modified.
     created_on = (
-        datastore_services.DateTimeProperty(indexed=True, required=True))
-    # When this entity was last updated. This value should only be modified by
-    # the update_timestamps method.
+        datastore_services.DateTimeProperty(required=True, indexed=True))
```

Comment:
Shouldn't we add auto_now=True here and for last_updated as in the doc or does the pre_put_hook takes care of it?

Reply:
`pre_put_hook` handles it, the behaviour of `auto_now` is weird and go against a few assumptions that we have in our codebase.

Reply:
Ok thanks.